src/ContainerResource.php

```diff
@@ -147,19 +184,33 @@ public function isProtected(): bool
         return $this->protected;
     }
 
+    /**
+     * Check whether the resource is lazy
+     *
+     * @return  boolean
+     *
+     * @since   __DEPLOY_VERSION__
+     */
+    public function isLazy(): bool
+    {
+        return $this->lazy;
+    }
+
     /**
      * Get an instance of the resource
      *
      * If a factory was provided, the resource is created and - if it is a shared resource - cached internally.
      * If the resource was provided directly, that resource is returned.
      *
+     * @param   boolean  $noLazy  Not to use the lazy proxy
+     *
      * @return  mixed
      *
      * @since   2.0.0
      */
-    public function getInstance()
+    public function getInstance(bool $noLazy = false)
```

Comment:
For bc, this has to be true. In a new major we can change then to false.

Reply:
@laoneo, fixed.